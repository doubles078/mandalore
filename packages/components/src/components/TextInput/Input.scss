.TextInput {
  font-family: $typography-main;

  &__error {
    display: block;
    font-size: 0.75rem;
    margin-bottom: $componentSpacing-s;
  }

  &__label {
    color: $colors-prestigeBlue;
    display: block;
    font-size: 0.75rem;
    margin-bottom: $componentSpacing-xxs;
  }

  &__input {
    border-radius: $inputs-borderRadius;
    border: 1px solid $colors-prestigeBlue;
    font-size: 0.875rem;
    height: $inputs-height;
    margin-bottom: $componentSpacing-s;
    padding: 0 1rem;
    width: 100%;
    transition: all 300ms ease-in-out;

    &:focus,
    &:active {
      border: 1px solid $colors-clearChill;
      box-shadow: 0 4px 12px 0 $colors-frenchSkyBlue;
      outline: none;
    }
  }

  &--error {
    .TextInput__error {
      color: $colors-watermelon;
    }

    .TextInput__label {
      color: $colors-watermelon;
    }

    .TextInput__input {
      border-color: $colors-watermelon;
      color: $colors-watermelon;
      margin-bottom: $componentSpacing-xxs;
    }
  }

  &--disabled {
    .TextInput__label {
      opacity: 0.4;
    }

    .TextInput__input {
      background: $colors-prestigeBlue;
      border: 1px solid $colors-prestigeBlue;
    }
  }
}
