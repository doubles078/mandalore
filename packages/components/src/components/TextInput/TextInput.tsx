import classNames from "classnames";
import React from "react";
import { ValidationMessage } from "../";

export type TextInputProps = {
  disabled?: boolean;
  error?: any;
  id: string;
  input?: any;
  label: string;
  name?: string;
  value?: string;
  warning?: any;
};

export const TextInput = ({
  disabled,
  label,
  id,
  input,
  error,
  value,
  warning,
  ...other
}: TextInputProps) => {
  const inputClass = classNames({
    TextInput: true,
    "TextInput--disabled": disabled,
    "TextInput--error": error,
    "TextInput--warning": warning
  });
  const labelClass = classNames({
    TextInput__label: true
  });
  return (
    <div className={inputClass}>
      <label className={labelClass} htmlFor={id}>
        {label}
      </label>
      {input ? (
        input
      ) : (
        <input
          className="TextInput__input"
          disabled={disabled}
          id={id}
          value={value}
          {...other}
        />
      )}
      {error && <span className="TextInput__error">{error}</span>}
      {warning && (
        <ValidationMessage passive={true}>{warning}</ValidationMessage>
      )}
    </div>
  );
};
