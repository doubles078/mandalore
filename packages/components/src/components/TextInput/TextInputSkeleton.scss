.TextInputSkeleton {
  border-radius: $inputs-borderRadius;
  height: 1rem;
  margin-bottom: $input-margin-bottom;
  min-width: 400px;
  @include skeleton;
}
