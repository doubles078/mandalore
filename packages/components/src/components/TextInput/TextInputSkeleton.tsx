import React from "react";

export const TextInputSkeleton = () => {
  return <div className={`TextInputSkeleton`} />;
};
