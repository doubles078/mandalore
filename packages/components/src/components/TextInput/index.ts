export { MaskedTextInput } from "./MaskedTextInput";
export { TextInput, TextInputProps } from "./TextInput";
export { TextInputSkeleton } from "./TextInputSkeleton";
