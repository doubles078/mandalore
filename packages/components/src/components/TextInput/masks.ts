export const Masks = {
  CanadaPostalCode: [/[A-Z]/i, /\d/, /[A-Z]/i, " ", /\d/, /[A-Z]/i, /\d/],
  DateOfBirth: [/\d/, /\d/, "/", /\d/, /\d/, "/", /\d/, /\d/, /\d/, /\d/],
  Phone: [
    "(",
    /[1-9]/,
    /\d/,
    /\d/,
    ")",
    " ",
    /\d/,
    /\d/,
    /\d/,
    "-",
    /\d/,
    /\d/,
    /\d/,
    /\d/
  ],
  USZipCode: [/\d/, /\d/, /\d/, /\d/, /\d/]
};
