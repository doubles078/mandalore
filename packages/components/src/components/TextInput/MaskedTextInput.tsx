import React from "react";
import MaskedInput from "react-text-mask";

import { Masks } from "./masks";
import { TextInput, TextInputProps } from "./TextInput";

export interface IMaskedTextInput extends TextInputProps {
  mask:
    | "DateOfBirth"
    | "CanadaPostalCode"
    | "USZipCode"
    | "Phone"
    | Array<string | RegExp>;
}

export const MaskedTextInput: React.SFC<IMaskedTextInput> = ({
  mask,
  ...other
}) => {
  const maskedInput = (
    <MaskedInput
      className="TextInput__input"
      mask={typeof mask === "string" ? Masks[mask] : mask}
      {...other}
    />
  );
  return <TextInput input={maskedInput} {...other} />;
};
