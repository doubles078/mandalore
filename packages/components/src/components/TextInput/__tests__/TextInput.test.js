import React from "react";
import { TextInput } from "../TextInput";
import renderer from "react-test-renderer";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

test("TextInput renders with a label", () => {
  const component = renderer.create(<TextInput label="A label" id="test" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const textInput = shallow(<TextInput label="A label" id="test" />);

  expect(textInput.find("label").text()).toEqual("A label");
});
