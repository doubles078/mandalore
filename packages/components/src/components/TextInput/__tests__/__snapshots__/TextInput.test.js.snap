// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders with a label 1`] = `
<div
  className="TextInput"
>
  <label
    className="TextInput__label"
    htmlFor="test"
  >
    A label
  </label>
  <input
    className="TextInput__input"
    id="test"
  />
</div>
`;
